File: Tareas/12/Código JAVA/tarea12.java
/*	
	Nombre: Leslie Estefanía Pérez Cerda
	Matrícula: 1737535
	Hora: Jueves V4-V6
*/

import javax.swing.JOptionPane;
public class tarea12{
	public static void main (String[] args){
		String entrada;
		String serie = "";
		int n,i,j,a;

		entrada = JOptionPane.showInputDialog("Ingrese el numero de terminos: ");
		n = Integer.parseInt(entrada);

		for (i=0; i<n; i++){
			if(i%2==0){
				a = i;
				j = 0;
				while(i<n){
					if( j<a ){
						serie = serie+a+", ";
						i++;
						j++;
						JOptionPane.showMessageDialog(null,serie);

					}else{JOptionPane.showMessageDialog(null,"");}
				}
			}else{
				serie = serie+" -"+i+", ";
			}
		} //Fin del ciclo for
		JOptionPane.showMessageDialog(null,serie);
	} //Fin de función main
} //Fin de clase tarea12

File: Tareas/12/Código RUBY/tarea12.rb
=begin
	Nombre: Leslie Estefanía Pérez Cerda
	Matrícula: 1737535
	Hora: Jueves V4-V6
=end

print("Ingrese el numero de terminos: ")
n = gets().to_i

serie = ""
y = 1
x = 0
z = 1
a = 2

if(n>0)then
	serie = serie + "-1";

	for i in(1...n)
		serie = serie + ", "
		if (z == 0)then
			serie = serie + (a*-1).to_s

			z = 1
			x = 0
			a = a + 1
		elsif(z == 1)then
			serie = serie + a.to_s

			if (x<y)then
				x = x + 1
			else
				z = 0
				a = a + 1
				y = y + 1
			end
		end
	end
end

print(serie)